### ISSUE TITLE
linter: `react/react-compiler` CapitalizedCalls false positive on built-in globals (e.g. `BigInt()`)

### ISSUE THREAD

**Issue by Suhaibinator** [oxc-project/oxc#24405]
## What version of Oxlint are you using?

1.73.0

## What command did you run?

```
oxlint repro.tsx
```

with this `.oxlintrc.json`:

```json
{
  "plugins": ["react"],
  "rules": { "react/react-compiler": "error" }
}
```

(equivalently: `oxlint --react-plugin --deny react/react-compiler repro.tsx`)

## What happened?

The `react/react-compiler` rule's `CapitalizedCalls` check reports a false positive on calls to built-in global functions that happen to start with a capital letter — most notably `BigInt(...)`. It assumes any capitalized call is a React component being incorrectly invoked as a function.

`repro.tsx`:

```tsx
function Counter({ value }: { value: string }) {
  const id = BigInt(value);      // <-- flagged
  return <div>{id.toString()}</div>;
}
export default Counter;
```

Output:

```
repro.tsx:2:14: error react(react-compiler): CapitalizedCalls: Capitalized functions are reserved for components, which must be invoked with JSX. If this is a component, render it with JSX. Otherwise, ensure that it has no hook calls and rename it to begin with a lowercase letter. Alternatively, if you know for a fact that this function is not a component, you can allowlist it via the compiler config help: BigInt may be a component
```

Note the diagnostic even says **"BigInt may be a component"**.

The same misfire applies to other capitalized built-ins when called as functions (e.g. `Number(x)`, `Boolean(x)`, `String(x)`, `Symbol(x)`, `Array(x)`, `Object(x)`, `Date(x)`).

Two problems compound this:

1. **`BigInt` (and the other built-in globals above) should be recognized as non-components** and excluded from `CapitalizedCalls` out of the box. Meta's React Compiler does not flag these.
2. **The suggested remedy is not actually available in Oxlint.** The diagnostic says to *"allowlist it via the compiler config"*, but Oxlint's `ReactCompilerConfig` only exposes `reportAllBailouts` — there is no allowlist / `environment` option, so there is no way to silence this short of a per-line `// eslint-disable-next-line react/react-compiler` on every call site.

## What did you expect to happen?

`BigInt(value)` (and other calls to well-known capitalized built-in globals) should not trigger `CapitalizedCalls`. No diagnostic should be produced for the repro above.

## Impact

In a codebase that uses `BigInt()` for 64-bit ID conversion, this fires on essentially every page/component that converts an ID — in ours it produced 47 false positives across ~40 files, which makes `react/react-compiler` unusable as an `error`-level rule despite the rest of the rule working well (it correctly catches conditional hooks, setState-in-render, etc.).

## Environment

- oxlint 1.73.0
- Platform: macOS (darwin, arm64)

**Comment by durationextender** [comment]
@camc314 can i work on this issue?

**Comment by camc314** [comment]
@Boshen what's our plan for these issues? do we fix it upstream?

**Comment by durationextender** [comment]
@camc314 i made a pr with a potential fix.

**Comment by feychenie** [comment]
As of oxlint v1.81.0, it is still an issue with namespaced internals, like `Intl.DateTimeFormat()`
